### PR TITLE
MCR-2843 Harmonize periodically performed background tasks

### PR DESCRIPTION
**Pull request by toKrause**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2843).

### CODE REVIEW

**Comment by sebhofmann** [comment]
We already have the [Cronjob-API](https://www.mycore.de/documentation/basics/basics_cronjob/). I just did not have the time to migrate the old Code to the new API.

**Comment by toKrause** [comment]
> We already have the [Cronjob-API](https://www.mycore.de/documentation/basics/basics_cronjob/). I just did not have the time to migrate the old Code to the new API.

These jobs are periodic jobs not cron jobs, i.e., they are scheduled to run periodically, not at a specified time (although that might not be an important difference in real world scenarios.). The `Cronjob`-affix that the classes had was a bit misleading (presumably the names were given, before `MCRCronjob` was created).

I still think that this PR is a step forward. Specifically, 

- later conversion to cron jobs is still possible
- later conversion to cron jobs should be easier, since all the code concerning scheduling is now removed from the concrete job implmentations

**Review by sebhofmann — APPROVED** [review]

File: mycore-mods/src/main/java/org/mycore/mods/MCRMODSEmbargoReleaseCronjob.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import org.mycore.datamodel.metadata.MCRObject;
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.mcr.cronjob.MCRCronjob;
+import org.mycore.solr.MCRSolrClientFactory;
+import org.mycore.util.concurrent.MCRFixedUserCallable;
+
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
```
Comment (yagee-de) [inline mycore-mods/src/main/java/org/mycore/mods/MCRMODSEmbargoReleaseCronjob.java]: fix import order

File: mycore-pi/src/main/java/org/mycore/pi/urn/MCRDNBURNRegistrationCheckCronjob.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import org.mycore.mcr.cronjob.MCRCronjob;
+import org.mycore.pi.MCRPIManager;
+import org.mycore.pi.MCRPIServiceManager;
+import org.mycore.pi.backend.MCRPI;
+import org.mycore.pi.exceptions.MCRIdentifierUnresolvableException;
+import org.mycore.util.concurrent.MCRFixedUserCallable;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.Optional;
```
Comment (yagee-de) [inline mycore-pi/src/main/java/org/mycore/pi/urn/MCRDNBURNRegistrationCheckCronjob.java]: fix import order

File: mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRURNGranularRESTRegistrationCronjob.java [hunk 3]
```diff
@@ -18,60 +18,75 @@
...
+import org.mycore.mcr.cronjob.MCRCronjob;
 import org.mycore.pi.MCRPIManager;
+import org.mycore.pi.MCRPIRegistrationInfo;
 import org.mycore.pi.urn.MCRDNBURN;
 
-import jakarta.persistence.EntityTransaction;
-import jakarta.persistence.RollbackException;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
```
Comment (yagee-de) [inline mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRURNGranularRESTRegistrationCronjob.java]: fix import order

File: mycore-pi/src/test/java/org/mycore/pi/urn/rest/MCRURNGranularRESTServiceTest.java [hunk 4]
```diff
@@ -53,6 +41,17 @@
...
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
```
Comment (yagee-de) [inline mycore-pi/src/test/java/org/mycore/pi/urn/rest/MCRURNGranularRESTServiceTest.java]: fix import order

**Review by yagee-de — DISMISSED** [review]
more line removed than added. Good job!
Does it require a migration?
Import order should be like:
```
java.*
<blank line>
javax.*
<blank line>
org.*
<blank line>
com.*
<blank line>
<all other imports>
```

**Comment by toKrause** [comment]
> Does it require a migration?

Yes an no.

Generally, a contributeion to `MCR.Startup.Class` has to be removed and entries like `CR.Cronjob.Jobs.XXX=...` and `CR.Cronjob.Jobs.XXX.Cron=...` have to be added.

For `MCRDNBURNRegistrationCheckCronjob` (formerly `MCRPICronJob`) the necessary changes are included in this PR.

For `MCRMODSEmbargoReleaseCronjob` (formerly `MCRMODSEmbargoCronjob`) the necessary change are included in https://github.com/MyCoRe-Org/mir/pull/844. Non-MIR-User would have to make the same changes themselves. A corresponding entry in `deprecated.properties` is included in this PR.

For `MCRURNGranularRESTRegistrationCronjob` (formerly `MCRURNGranularRESTRegistrationStarter`) similar changes would have to made by users of this job. This job isn't configured by default in either MyCoRe or MIR, so no changes are part of any PR.

**Review by kkrebs — APPROVED** [review]

**Pull request merged**